### DISCUSSION TITLE
[Feature] Public share via link

### DISCUSSION THREAD

**Discussion by cibernox** [immich-app/immich#3220] [Feature Request]
### The feature

Much like google photos allows you to create a link that allows to share a photo or album with anyone that has that link, it would be nice to share photos with family members that don't have an account in my immich instance.

For that, we'd need some config option to specify the public facing URL of the immich instance (e.g. `photos.myhomelab.com`) and some long unguessable token that gives however has it access to an album or resource.

I didn't find any reference for this feature among the requested features, I may have not used the right search terms since it seems a quite common feature (at least I use it often)

### Platform

- [X] Server
- [X] Web
- [X] Mobile

**Comment by alextran1502** [comment]
You can already create public shared link to share an entire album or any group of photos/videos

**Reply by alextran1502** [reply]
<img width="2056" alt="image" src="https://github.com/immich-app/immich/assets/27055614/d4ec173c-dc3e-492b-ba92-20356e15a17f">

<img width="423" alt="image" src="https://github.com/immich-app/immich/assets/27055614/819be45f-167a-4160-8f7a-a799a0f9f748">

<img width="427" alt="image" src="https://github.com/immich-app/immich/assets/27055614/6e53f96f-eb03-4186-8eaf-1e710921dcf2">

https://demo.immich.app/share/dKCymSJnA5niMfVcx5nL-32d61mecmzKwj6um84teXOfoh9Esb3hsMbKUWJ0QYH3XvY

**Reply by cibernox** [reply]
Oh, I didn’t notice it because that feature doesn’t seem to be available on the iOS app. I almost always use this feature on the go.

**Reply by mirisbowring** [reply]
Will this also be possible for single images?

**Reply by jrasm91** [reply]
This is possible for single images.

**Reply by mirisbowring** [reply]
But how? Did not find the correct place/ button i think

**Reply by jrasm91** [reply]
You have to use the multi select feature from the main timeline or album view to pick one and then hit the share button. Basically what Alex showed, but only selecting one photo.

**Reply by traktuner** [reply]
On iOS, when I select one or more pictures/videos and tap the share button, it downloads the whole asset (for example, I wanted to share a 4k video and it downloaded 600MB), then it offered me to share the full video with other apps.
I did not find a way - at least on iOS - to click a "create share link" button. So in my opinion that feature is not implemented yet on iOS.
In order to preserve space on other people's phones I woud like to only share one or more assets via link.

Please guide me how to do that if it actually IS implemented :) 

Thank you
